app/models/cart.rb

```diff
@@ -130,6 +130,10 @@ def add_items_to_checkout(order)
     order
   end
 
+  def shopping_cart_items_equal_order_items?(order)
+    number_of_shopping_cart_items == order.order_items.length
```

Comment:
`order.order_items.map(&:variant_id).sort == cart.cart_items.map(&:variant_id).sort`

This would be more accurate.   Also it might be best to submit the amount shown to the user and compare it to the amount based in the DB currently.  Example => if a sale ends or the price of the item changes.